components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java

```diff
@@ -1059,6 +1070,19 @@ public static boolean isSelfSignUpEPAvailable() {
         return false;
     }
 
+    /**
+     * Returns whether the email based username is enabled or not.
+     *
+     * @return true if the EnableEmailUserName. False if it doesnt
+     */
+    public static boolean isEmailUsernameEnabled() {
+        String enableEmailUsernameProperty = ServerConfiguration.getInstance().getFirstProperty(ENABLE_EMAIL_USERNAME);
+        if (StringUtils.isNotBlank(enableEmailUsernameProperty)) {
```

Comment:
Since parseBoolean logic is "return ((s != null) && s.equalsIgnoreCase("true"));" this isNotBlank is redundent.

Reply:
Fixed in 1700a907fb8dbfb0b3c0a908a8c62b3df2113266